themes/bootstrap3/js/common.js

```diff
@@ -687,25 +687,3 @@ $(function commonDocReady() {
     $("link[media='print']").attr("media", "all");
   }
 });
-
-$(function searchFormResetHandler() {
-  const queryInput = $('#searchForm_lookfor');
-  const resetButton = $('#searchForm-reset');
-
-  let query = queryInput.val();
-  if (query !== '') {
-    resetButton.show();
-    queryInput.focus().val('').val(query);
```

Comment:
There are some bug fixes currently being worked on in #3064 that impact this code... but it looks like you've removed or significantly rewritten it as part of the refactoring here. I'm going to wait until that PR is finished before doing final testing and merging of this one, since it seems better to fix existing bugs before adding new functionality on top of them. If you have a moment to look at #3064 and comment on it (if it hasn't been merged yet) or reconcile the fixes with this PR (if it has been completed by the time you read it), I'd appreciate the help!

Reply:
It looks like this was moved to the [_onChange handler for the keyboard](https://github.com/vufind-org/vufind/pull/2952/files/90efef862a9b165e167aa9951ea56ac9fbc9ec00#diff-fbacfd93e0c0eb3a264f477eff397a1c48a952fc877a3f674e9aee6ebf62b103R44). It looks to me like this should work even when the keyboard is disabled, but I haven't tested it myself.

Reply:
I am not sure what the purpose of #3064 is. Was it introduced so that "Loading..." is not shown when the reset button is used or when the page is loaded initially?

In this PR whenever a change is made to the input by using the actual keyboard, the virtual keyboard or the reset button the _onChange function is used to update the state for all the elements. And 

>  [_onChange handler for the keyboard](https://github.com/vufind-org/vufind/pull/2952/files/90efef862a9b165e167aa9951ea56ac9fbc9ec00#diff-fbacfd93e0c0eb3a264f477eff397a1c48a952fc877a3f674e9aee6ebf62b103R44)

is used to trigger the autocomplete.

Reply:
The behavior you describe does not occur on this PR at least in my setup. But the searchbox is not auto-focused also. 

Is the auto-focus a wanted but it should just not open the autocomplete? In that case, I could add `_textInput.focus();` to the `_onChange` function and it still works.

Reply:
I think it's generally better to avoid auto-focus for accessibility reasons, so if it's not auto-focusing at present, I don't think we need to make changes to make it do so. :-) Thanks for looking at this!